merkle/objhasher/objhasher.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objhasher
+
+import (
+	"crypto/sha256" // Use SHA256 to match ObjectHash.
+
+	"github.com/benlaurie/objecthash/go/objecthash"
+)
+
+// ObjectHasher uses ObjectHash to compute leaf hashes.
+var ObjectHasher = objhasher{}
+
+// ObjectHash does not use `1` as any of its type prefixes,
+// preserving domain separation.
+const nodeHashPrefix = 1
+
+type objhasher struct{}
+
+// HashEmpty returns the hash of an empty element for the tree
+func (t objhasher) HashEmpty() []byte {
+	return sha256.New().Sum(nil)
+}
+
+// HashLeaf returns the object hash of leaf, which must be a JSON object.
+func (t objhasher) HashLeaf(leaf []byte) []byte {
+	hash := objecthash.CommonJSONHash(string(leaf))
```

Comment:
No leaf prefix. A couple reasons: 
1. The prefix establishes domain separation, but ObjectHash has it's own set of prefixes, none of which overlap with 0x01 for nodes, so we already get domain separation. 
2. It's difficult to add a prefix to the output of another hash function, unless you want to do a double hash - something like `H(0x00 || ObjectHash(m))`

Unless I misunderstand, a "TreeHasher" is a HashStrategy. Should it be added to the set of hash strategies? I find this requirement slightly unfortunate because the only change here is how leaf hashes are calculated. HashEmpty and HashChildren are exactly the same as rfc6962. If the server didn't [overwrite](#398) a the MerkleLeaf hash supplied by the client, no further changes would be needed, but as it stands, we're on track add a new hash strategy for each new way of computing leaf nodes. Happy to do it though.